### ISSUE TITLE
Latest version does not support Angular 15 which results in dependency conflicts for ngx-flow when updating to Angular 15.

### ISSUE THREAD

**Issue by sukantoghsh** [flowjs/ngx-flow#97]
What would be the best approach to solve this ?

**Comment by Quentigus** [comment]
I need Angular 15 support too :+1:

**Comment by evilaliv3** [comment]
@sukantoghsh / @Quentigus : which is your experience with Angular15? Did you encounter any incompatibility?

@MartinNuc : which are you evaluations?

Are you aware of any issue with angularjs?

**Comment by evilaliv3** [comment]
Hello @MartinNuc, i just wanted to reming you about this ticket in case you missed to see my comment.

@sukantoghsh @Quentigus: do you confirm that all is working perfectly on the current Angular?

**Comment by evilaliv3** [comment]
Thank you.

Did you encounter any error testing it?

**Comment by StavNoyAkur8** [comment]
Our project has been working with Angular 15 and ngx-flow since January using legacy-peer-deps. No issues.
Thanks @Quentigus for the tip on NPM overrides

**Comment by rmhubley** [comment]
What about Angular 16?  I am getting this error:

```
Error: src/app/workbench/workbench.module.ts:53:5 - error NG6002: 'NgxFlowModule' does not appear to be an NgModule class.

53     NgxFlowModule,
       ~~~~~~~~~~~~~

  node_modules/@flowjs/ngx-flow/lib/ngx-flow.module.d.ts:2:22
    2 export declare class NgxFlowModule {
                           ~~~~~~~~~~~~~
    This likely means that the library (@flowjs/ngx-flow) which declares NgxFlowModule is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to be compatible with Ivy.
```

**Comment by Quentigus** [comment]
Is it possible de release a new version with a more permissive `peerDependencies`  ?

For exemple :
```
"peerDependencies": {
    "@angular/common": ">=14.0.0",
    "@angular/core": ">=14.0.0",
    "@flowjs/flow.js": "^2.13.0"
},
```

Actual values ("^14.0.0-") seems to be invalid semver values

**Comment by CosmoFruit** [comment]
fixed https://github.com/flowjs/ngx-flow/pull/105

**Comment by CosmoFruit** [comment]
@MartinNuc Hi, can you publish 0.7.0 release?

**Comment by evilaliv3** [comment]
Thank you for the reminder.


Are you all confident that the version 0.7.0 is working properly?
Can someone confirm this and eventually ensure that the dependencies included in 0.7.0 are up to date and tests working?

I'm not following the project so much but i may try to do the publishing if you could confirm this.

Is there anyone that would like to support the maintainance of the project?

**Comment by CosmoFruit** [comment]
> Is there anyone that would like to support the maintainance of the project?

@evilaliv3 i can support the maintainance of the project, bcz we use it in our enterprise, and we need ability update dependencies

**Comment by evilaliv3** [comment]
Thank you @CosmoFruit 

Would you like to make a pull request in relation to the up to date changes that you consider are needed?
Viewing some contributions would help us all evaluate who could move on with the maintainance of the project on the long run.

**Comment by CosmoFruit** [comment]
@evilaliv3 ready, all is work fine, updated only compatible deps. will waiting actual release in npm =)

**Comment by CosmoFruit** [comment]
@evilaliv3 also, can you grant acess to this repo? it would be great to put things in order and close the outdated issues

**Comment by CosmoFruit** [comment]
@evilaliv3  any updates?

**Comment by sebastiencl** [comment]
Any news about the npm publication ? I'm stuck with my overrides since more than a year.
0.7.0 would be the perfect gift for christmas :)

**Comment by sebastiencl** [comment]
@MartinNuc are you here ?

And happy new year of course ;)

**Comment by evilaliv3** [comment]
O Saturday or Sunday I will try to see if I still have the permission to
make the release. I've also to remember how :)

**Comment by AidasK** [comment]
I have also added @CosmoFruit to the maintainers, maybe he still wants to help :))

**Comment by Quentigus** [comment]
Hello,
You can add me too, I will try to help 👍

**Comment by evilaliv3** [comment]
Ive built and published the 0.7.0 on npm.

Would you please test on your projects if the publishing was correct? Thank you

**Comment by sebastiencl** [comment]
It works great. Thanks a lot !

**Comment by evilaliv3** [comment]
Is anyone able and interested in supporting the update of the library to angular 17?

**Comment by CosmoFruit** [comment]
great! thx.


> Is anyone able and interested in supporting the update of the library to angular 17?

i will planning  do it at this spring.  when we will migrated on angular 17

**Comment by CosmoFruit** [comment]
this issue can be closed

**Comment by CosmoFruit** [comment]
> I have also added @CosmoFruit to the maintainers, maybe he still wants to help :))

can you repeat invite plz ? =)

**Comment by evilaliv3** [comment]
Is there anyone interested supporting updating the library for Angular 17?

**Comment by CosmoFruit** [comment]
@evilaliv3 @AidasK can you add me to npm org?  i build patch, and want try publish

**Comment by AidasK** [comment]
@CosmoFruit what's your npmjs username?

**Comment by CosmoFruit** [comment]
> @CosmoFruit what's your npmjs username?

https://www.npmjs.com/~cfruit

**Comment by AidasK** [comment]
@evilaliv3 can you add cfruit? It looks like only the owner can do it

**Comment by evilaliv3** [comment]
I can definitely do it, but ar the moment @CosmoFruit we have already a new package fully supporting Angular 16.

Ars you able to prepare a pull request to add support for Angular 17 ?

**Comment by Quentigus** [comment]
> i made litle fix https://github.com/flowjs/ngx-flow/pull/107 for upgrade DX (avoid --legacy-peer-deps)

0.7.1 should be published asap :)

**Comment by CosmoFruit** [comment]
@evilaliv3 @AidasK @MartinNuc can you publish 0.7.1 ?

**Comment by Quentigus** [comment]
I will need, for my work, an Angular 18 version, I can take care of this compatibility upon release. For v17, I don't have much time at the moment, I'll try but I can't promise anything.

**Comment by Quentigus** [comment]
I've installed the 0.7.1 version inside my project and seems to be broken :(
This is not a compiled bundle (left = 0.7.0 - right = 0.7.1) :

![image](https://github.com/flowjs/ngx-flow/assets/5646533/6d1ac662-8d17-4dd2-91f7-bb5baba04b88)

Error in project (only with 0.7.1 :
![image](https://github.com/flowjs/ngx-flow/assets/5646533/d86ff2a8-9677-4167-ba3e-78ac91fc7c5a)

**Comment by evilaliv3** [comment]
Ah, this is probably my fault.

I just run npm publish.

Do you know the commands that i need to run to have a compiled bundle?

**Comment by evilaliv3** [comment]
Ok unpublished, we will be able to publish the same corrected version after 24 hours

**Comment by Quentigus** [comment]
Can you add me to the npm org, please ?
https://www.npmjs.com/~quentigus

I can publish the 0.7.1 and migrate to Angular 17

**Comment by evilaliv3** [comment]
Sure, thank you.

I just tried to re-publish the 0.7.1 but actually npm prevents it.
I think you will need to publish a 0.7.2

**Comment by Quentigus** [comment]
The invitation is for another project : https://www.npmjs.com/package/ngx-flow

**Comment by Quentigus** [comment]
Can you add me on https://www.npmjs.com/package/@flowjs/ngx-flow ?

**Comment by evilaliv3** [comment]
@Quentigus : oh sorry, that was a test one. i tried to add you now, have you received the invite?

In the meantime due to the rules of npm that correctly prevents to publish a version already published, i've published a 0.7.2 that fixes the 0.7.1 release

**Comment by Quentigus** [comment]
The first , yes, but not the second.
